ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/JsonCallParameter.java

```diff
@@ -26,68 +26,339 @@
         value,
-        Optional.ofNullable(input != null ? input : data).orElse(null),
-        Optional.ofNullable(accessList),
-        Optional.ofNullable(maxFeePerBlobGas),
-        Optional.ofNullable(blobVersionedHashes));
+        payload,
+        accessList,
+        maxFeePerBlobGas,
+        blobVersionedHashes);
 
-    if (input != null && data != null && !input.equals(data)) {
-      throw new IllegalArgumentException("Only one of 'input' or 'data' should be provided");
-    }
-
-    this.strict = Optional.ofNullable(strict);
+    this.strict = strict;
   }
 
+  /**
+   * Returns whether the call should be executed in strict mode.
+   *
+   * @return Optional strict mode flag
+   */
   public Optional<Boolean> isMaybeStrict() {
     return strict;
   }
 
-  @JsonAnySetter
-  public void logUnknownProperties(final String key, final Object value) {
-    LOG.debug(
-        "unknown property - {} with value - {} and type - {} caught during serialization",
-        key,
-        value,
-        value.getClass());
+  /**
+   * Builder for {@link JsonCallParameter}. Used by Jackson to deserialize {@code
+   * JsonCallParameter}.
+   */
+  public static final class JsonCallParameterBuilder {
+    private Optional<Boolean> strict = Optional.empty();
+    private Address from;
+    private Address to;
+    private long gas = -1;
+    private Optional<Wei> maxPriorityFeePerGas = Optional.empty();
+    private Optional<Wei> maxFeePerGas = Optional.empty();
+    private Optional<Wei> maxFeePerBlobGas = Optional.empty();
+    private Wei gasPrice;
+    private Wei value;
+    private Bytes data;
+    private Bytes input;
+    private Optional<List<AccessListEntry>> accessList = Optional.empty();
+    private Optional<List<VersionedHash>> blobVersionedHashes = Optional.empty();
+
+    /** Default constructor. */
+    public JsonCallParameterBuilder() {}
+
+    /**
+     * Sets the strict mode for the {@link JsonCallParameter}. If strict mode is enabled, the call
+     * will be executed with stricter validation rules. This is optional and defaults to not being
+     * in strict mode if not specified.
+     *
+     * @param strict the strict mode flag, can be {@code null} to indicate the absence of a strict
+     *     mode preference
+     * @return the {@link JsonCallParameterBuilder} instance for chaining
+     */
+    public JsonCallParameterBuilder withStrict(final Boolean strict) {
+      this.strict = Optional.ofNullable(strict);
+      return this;
+    }
+
+    /**
+     * Sets the "from" address for the {@link JsonCallParameter}. This address represents the sender
+     * of the call.
+     *
+     * @param from the sender's address
+     * @return the {@link JsonCallParameterBuilder} instance for chaining
+     */
+    public JsonCallParameterBuilder withFrom(final Address from) {
+      this.from = from;
+      return this;
+    }
+
+    /**
+     * Sets the "to" address for the {@link JsonCallParameter}. This address represents the
```

Comment:
should we mention that this can be null, which means that this is a contract creation?